### PR TITLE
Any elements may now be added to a filter tag in filters.xml files and p...

### PR DESCRIPTION
**Pull request by pmbohm**
...ersist while a filter is active via the gui

### CODE REVIEW

File: src/main/java/au/org/emii/geoserver/extensions/filters/layer/data/Filter.java [hunk 1]
```diff
@@ -104,8 +114,10 @@ public Filter merge(Filter other) {
         filter.setVisualised(coalesce(right.getVisualised(), getVisualised()));
         filter.setEnabled(coalesce(right.getEnabled(), getEnabled()));
         filter.setExcludedFromDownload(coalesce(right.getExcludedFromDownload(), getExcludedFromDownload()));
+        filter.extras = right.extras; // only ever take the saved xml value
 
         return filter;
+
```
Comment (danfruehauf) [inline src/main/java/au/org/emii/geoserver/extensions/filters/layer/data/Filter.java]: Redundant empty line here.

File: src/main/java/au/org/emii/geoserver/extensions/filters/layer/data/Filter.java [hunk 2]
```diff
@@ -8,7 +8,7 @@
 package au.org.emii.geoserver.extensions.filters.layer.data;
 
 import org.apache.commons.lang.StringUtils;
-
+import java.util.*;
```
Comment (danfruehauf) [inline src/main/java/au/org/emii/geoserver/extensions/filters/layer/data/Filter.java]: Do we need to import `*` here?

File: src/test/java/au/org/emii/geoserver/extensions/filters/layer/data/FilterMergeTest.java [hunk 3]
```diff
@@ -29,6 +29,12 @@ public void setupLeftFilters() {
             filter.setLabel(String.format("left filter %s", i));
             leftFilters.add(filter);
         }
+
+        Filter filter = new Filter("TimeRange", "String");
+        filter.setLabel("timerange test");
+        filter.setExtrasField("ATESTFIELD", "thetestvalue");
+        leftFilters.add(filter);
+
```
Comment (danfruehauf) [inline src/test/java/au/org/emii/geoserver/extensions/filters/layer/data/FilterMergeTest.java]: Redundant blank line.

**Comment by danfruehauf** [comment]
@julian1 I'd appreciate if you could also have a look at this PR.

File: src/main/java/au/org/emii/geoserver/extensions/filters/layer/data/Filter.java (hunk 2 above)
Comment (pmbohm) [inline src/main/java/au/org/emii/geoserver/extensions/filters/layer/data/Filter.java]: I dont know?
The jar is only 68KB
Whats the downside?

Comment (danfruehauf) [inline src/main/java/au/org/emii/geoserver/extensions/filters/layer/data/Filter.java]: I'm not a java expert, but the best practice is to import just what you need I think.

**Comment by pmbohm** [comment]
@danfruehauf New whitespace changes now added

**Pull request merged**